src/main/java/com/minecolonies/coremod/entity/pathfinding/PathNavigate.java

```diff
@@ -82,20 +84,20 @@ public Path getPathToPos(final BlockPos pos)
     /**
      * Get the destination from the path.
      *
-     * @return the destionation position.
+     * @return the destination position.
      */
     public BlockPos getDestination()
     {
         return destination;
     }
 
-    //  Re-enable this if path shortcutting becomes a problem; then entities will move more rigidly along world grid
-//    @Override
-//    protected boolean isDirectPathBetweenPoints(final Vec3d Vec3d, final Vec3d Vec3d1, final int i, final int i1, final int i2)
-//    {
-//        //we don't use, so it doesn't matter
-//        return false;
-//    }
+    @Override
+    protected boolean isDirectPathBetweenPoints(final Vec3d start, final Vec3d end, int sizeX, int sizeY, int sizeZ)
+    {
+        // TODO improve road walking. This is better in some situations, but still not great.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)